[feature](inverted_index) add ik analyzer for chinese language

Ryan19929 opened the pull request:
### What problem does this PR solve?

Issue Number: close #35411 

Related PR: #xxx

Problem Summary:
This pull request adds support for the IK Tokenizer to the project. Key changes include migrating the ik tokenizer  from Java to C++, updating the inverted index parser to work with IK, and adding new test cases for the IK analyzer.
## IK Integration:
- `be/CMakeLists.txt`: Added installation of IK dict files to the output directory.
## Inverted Index Parser Updates:
- `be/src/olap/inverted_index_parser.cpp`: Added support for the PARSER_IK type in the `inverted_index_parser_type_to_string`, `get_inverted_index_parser_type_from_string` and `get_parser_mode_string_from_properties` functions.
- `be/src/olap/inverted_index_parser.h`: Defined PARSER_IK in the InvertedIndexParserType enum and added the corresponding string constant. 
- `be/src/olap/rowset/segment_v2/inverted_index/analyzer/analyzer.cpp`: Included the IKAnalyzer header and updated the create_analyzer function to handle the PARSER_IK type.
- `be/src/vec/functions/function_tokenize.cpp`: Update error message.
- `fe/fe-core/src/main/java/org/apache/doris/analysis/InvertedIndexUtil.java`: Added support for the IK analyzer.
## Test Cases
 
- `regression-test/suites/inverted_index_p0/test_ik_analyzer.groovy`: Added a new test suite for the IK analyzer, including table creation, data insertion, and query validation.
- `regression-test/suites/inverted_index_p0/test_tokenize.groovy`: Added test cases for tokenizing text using the IK parser.
- `regression-test/data/inverted_index_p0/test_ik_analyzer.out`: Added expected output for the IK analyzer test cases.
- `regression-test/data/inverted_index_p0/test_tokenize.out`: Added expected output for the tokenization test cases using the IK parser.

###  Release note

None

### Check List (For Author)

- Test <!-- At least one of them must be included. -->
    - [X] Regression test
    - [X] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason <!-- Add your reason?  -->

- Behavior changed:
    - [X] No.
    - [ ] Yes. <!-- Explain the behavior change -->

- Does this need documentation?
    - [ ] No.
    - [X] Yes. <!-- Add document PR link here. eg: https://github.com/apache/doris-website/pull/1214 -->

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label <!-- Add branch pick label that this PR should merge into -->

hello-stephen commented:
Thank you for your contribution to Apache Doris.
Don't know what should be done next? See [How to process your PR](https://cwiki.apache.org/confluence/display/DORIS/How+to+process+your+PR).

Please clearly describe your PR:
1. What problem was fixed (it's best to include specific error reporting information). How it was fixed.
2. Which behaviors were modified. What was the previous behavior, what is it now, why was it modified, and what possible impacts might there be.
3. What features were added. Why was this function added?
4. Which code was refactored and why was this part of the code refactored?
5. Which functions were optimized and what is the difference before and after the optimization?

zzzxl1993 commented:
run buildall

Ryan19929 commented:
run buildall

hello-stephen commented:
# BE UT Coverage Report
Increment line coverage `77.27% (1210/1566)` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/4f6aa52dfe583a6b32bb29f0e8b00bd7f1b626e2_4f6aa52dfe583a6b32bb29f0e8b00bd7f1b626e2/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/4f6aa52dfe583a6b32bb29f0e8b00bd7f1b626e2_4f6aa52dfe583a6b32bb29f0e8b00bd7f1b626e2/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 51.13% (13794/26978) |
| Line Coverage     | 40.47% (119477/295246) |
| Region Coverage   | 39.20% (60878/155288) |
| Branch Coverage   | 34.12% (30694/89956) |

Ryan19929 commented:
run buildall

Ryan19929 commented:
run buildall

hello-stephen commented:
# BE UT Coverage Report
Increment line coverage `77.27% (1210/1566)` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/ecf09e0752cd866c657ea555fd4ab17f18e4a330_ecf09e0752cd866c657ea555fd4ab17f18e4a330/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/ecf09e0752cd866c657ea555fd4ab17f18e4a330_ecf09e0752cd866c657ea555fd4ab17f18e4a330/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 51.15% (13800/26981) |
| Line Coverage     | 40.47% (119511/295287) |
| Region Coverage   | 39.22% (60894/155277) |
| Branch Coverage   | 34.13% (30699/89938) |

Ryan19929 commented:
run buildall

airborne12 commented:
run buildall

hello-stephen commented:
# BE UT Coverage Report
Increment line coverage `77.22% (1207/1563)` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/3992b672b1115ee9d16ca623b3f94c4629bb6290_3992b672b1115ee9d16ca623b3f94c4629bb6290/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/3992b672b1115ee9d16ca623b3f94c4629bb6290_3992b672b1115ee9d16ca623b3f94c4629bb6290/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 52.59% (14249/27095) |
| Line Coverage     | 41.34% (122811/297096) |
| Region Coverage   | 40.12% (62655/156177) |
| Branch Coverage   | 34.79% (31441/90376) |

Ryan19929 commented:
run p0

Ryan19929 committed:
[feature](inverted_index) add ik analyzer for chinese language

Migrate analysis-ik from Java to C++, implement basic tokenization functionality.

Ryan19929 commented:
run buildall

Ryan19929 commented:
run buildall

Ryan19929 commented:
run p0

xiaokang commented on be/src/olap/rowset/segment_v2/inverted_index/analyzer/analyzer.cpp:
```diff
@@ -73,6 +74,16 @@ std::unique_ptr<lucene::analysis::Analyzer> InvertedIndexAnalyzer::create_analyz
+        ik_analyzer->initDict(config::inverted_index_dict_path + "/ik");
+        auto mode = inverted_index_ctx->parser_mode;
+        if (mode == INVERTED_INDEX_PARSER_SMART) {
+            ik_analyzer->setMode(true);
```
true false mode is not scalable to add new ones.

xiaokang commented on be/CMakeLists.txt:
```diff
@@ -239,6 +239,11 @@ install(DIRECTORY
+
+install(DIRECTORY
+    ${BASE_DIR}/dict/ik
+    DESTINATION ${OUTPUT_DIR}/dict)
```
DESTINATION ${OUTPUT_DIR}/dict/ik

Ryan19929 replied:
> DESTINATION ${OUTPUT_DIR}/dict/ik

All dict files are in the `${BASE_DIR}/dict/ik` directory(`${BASE_DIR}/dict/ik/*.dict`). Setting `DESTINATION to ${OUTPUT_DIR}/dict/ik` would result in the path /dict/ik/ik/*.dict.

zzzxl1993 reviewed (approved):
LGTM

airborne12 reviewed (approved):
LGTM
